agent/utils/ephemeral_ports.go

```diff
@@ -89,22 +90,20 @@ func (pt *safePortTracker) GetLastAssignedHostPort() int {
 	return pt.lastAssignedHostPort
 }
 
-var dynamicHostPortRange = getDynamicHostPortRange
 var tracker safePortTracker
 
 // GetHostPortRange gets N contiguous host ports from the ephemeral host port range defined on the host.
-func GetHostPortRange(numberOfPorts int, protocol string) (string, error) {
+func GetHostPortRange(numberOfPorts int, protocol string, dynamicHostPortRange string) (string, error) {
 	portLock.Lock()
 	defer portLock.Unlock()
 
 	// get ephemeral port range, either default or if custom-defined
-	startHostPortRange, endHostPortRange, err := dynamicHostPortRange()
+	startHostPortRange, endHostPortRange, err := nat.ParsePortRangeToInt(dynamicHostPortRange) //dynamicHostPortRange()
 	if err != nil {
```

Comment:
this block won't be needed because you've already validated the cfg option before sending it here.